### PR TITLE
Add API test coverage for 277514@main

### PR DESCRIPTION
**Pull request by cdumez**
#### 101dfa3798e84da38cee58bf8fe468271739e7dc
<pre>
Add API test coverage for 277514@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=272780">https://bugs.webkit.org/show_bug.cgi?id=272780</a>

Reviewed by Per Arne Vollan.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isSuspended]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Tools/TestWebKitAPI/Tests/WebKit/PreferenceChanges.mm:
(TEST(WebKit, PreferenceChangesWithSuspendedProcess)):
(TEST(WebKit, GlobalPreferenceChangesUsingDefaultsWrite)):
(TEST(WebKit, GlobalPreferenceChangesUsingDefaultsWriteWithSuspendedProcess)):

Canonical link: <a href="https://commits.webkit.org/277615@main">https://commits.webkit.org/277615@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/268cc41c57f82fed61e072e1f93c1297dc49b8bb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48026 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27231 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50944 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50713 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44085 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50333 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33118 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24723 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39229 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48609 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24904 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41539 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20367 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22385 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/42715 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6077 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44370 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43149 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52606 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23072 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19427 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46558 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24339 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41689 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45426 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25139 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6829 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24060 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm [hunk 1]
```diff
@@ -145,6 +145,7 @@
 #if ENABLE(CFPREFS_DIRECT_MODE)
 void AuxiliaryProcessProxy::notifyPreferencesChanged(const String& domain, const String& key, const std::optional<String>& encodedValue)
 {
+    WTFLogAlways("CHRIS: notifyPreferencesChanged(%s, %s) m_isSuspended=%d", domain.utf8().data(), key.utf8().data(), m_isSuspended);
```
Comment (pvollan) [inline Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm]: Is this change intentional?

**Review by pvollan — APPROVED** [review]
R=me.

File: Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm (hunk 1 above)
Comment (cdumez) [reply Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm]: ahaha, that was my debug logging. Will drop.
